dev/releases/release_notes.py

```diff
@@ -316,6 +334,27 @@ def changes_overview(
         # finally copy over this new file to changelog.md
         os.rename(newfile, finalfile)
 
+def split_pr_into_changelog(prs: List):
+    childprlist = []
+    for pr in prs:
+        if has_label(pr, 'release notes: use body'):
+            mdstring = body_to_release_notes(pr).strip()
+            mdlines = mdstring.split('\r\n')
+            pattern = r'\{package: .*\}'
+            for line in mdlines:
+                if not '{package: ' in line:
+                    continue
+                mans = re.search(pattern, line)
+                packagestring = mans.group()[1:-1]
+                cpr = copy.deepcopy(pr)
+                mindex = line.find('{package:')
+                line = line[0:mindex]
+                cpr['labels'].append({'name': packagestring})
+                cpr['body'] = f'---\r\n## Release Notes\r\n{line}\r\n---'
+                childprlist.append(cpr)
+        prs.remove(pr)    
```

Comment:
Is removing elements from a list while iterating safe in Python? It isn't in many other languages...

In any case it might be cleaner to build a result list instead of removing from the input list, i.e. (pseudo code):
```
  result = []
  for pr in prs:
    if has_label(pr, 'release notes: use body'):
      ...
      for ..
         result.append(cpr)
    else:
      result.append(pr)
  return result
```